### PR TITLE
Remove locked werkzeug dependency due to CVE-2023-25577

### LINKED ISSUE #173: Current pinned Werkzeug version has a DoS Vulnerability (CVE-2023-25577)
I see in #123 there is a reason for an older werkzeug to be pinned, but that older version has a published CVE at this point.  Could we test/upgrade to a newer version?

https://www.cve.org/CVERecord?id=CVE-2023-25577

### PR DESCRIPTION
**Pull request by imichaelmoore**
Closes #173

### CODE REVIEW

**Comment by jonasvdd** [comment]
Hi @imichaelmoore, 

Thank you for creating both the issue and PR! 🤝 
I will manually test later on today whether the dash app hanging issue, as described in #123, still persists with more recent variants of `werkzeug`!
(This issue is rather hard to quantify in a test; as such the manual testing, maybe we should also make an issue about that 🙃)

**Comment by jvdd** [comment]
Instead of waiting for a new JupyterDash release (which can take quite a while) we could use the GitHub main as dependency until a new release is done?

**Comment by jvdd** [comment]
> Instead of waiting for a new JupyterDash release (which can take quite a while) we could use the GitHub main as dependency until a new release is done?

Installed their GitHub locally and did not observe any changes w.r.t. to #123 - the behavior is the same as in the current latest release. I'll now further experiment with werkzeug and try to write a test before we merge this #123

**Comment by jvdd** [comment]
If #176 passes and is merged, we can merge this one if the tests pass! :)

**Review by jvdd — APPROVED** [review]

**Comment by jvdd** [comment]
Tests seem to be passing :rocket: 
If you approve, you can merge this @jonasvdd

File: docs/sphinx/FAQ.rst [hunk 1]
```diff
@@ -66,6 +66,21 @@ The ``TraceUpdater`` class is a custom dash component that aids ``dcc.Graph`` co
...
+   <br>
+   <details>
+   <summary>
+      <a><b>My <code>FigureResampler.show_dash</code> keeps hanging (indefinitely) with the error message:<br>&nbsp;&nbsp;&nbsp; <code>OSError: Port already in use</code></b></a>
+   </summary>
+   <div style="margin-left:1em">
+
+Plotly-resampler its ``FigureResampler.show_dash`` method leverages the `jupyterdash <https://github.com/plotly/jupyter-dash>`_ toolkit to easily allow integration of dash apps in notebooks. However, there is a `known issue <https://github.com/plotly/jupyter-dash/pull/105>`_ with jupyterDash that causes the ``FigureResampler.show_dash`` method to hang when the port is already in use. In a future Pull-Request they will hopefully fix this issue.
+
+In the meantime, you can use the following workaround (if you do not care about the `Werkzeug security issue <https://github.com/predict-idlab/plotly-resampler/pull/174>`_): `pip install werkzeug==2.2.3`.
```
Comment (jvdd) [inline docs/sphinx/FAQ.rst:82]:
```suggestion
In the meantime, you can use the following workaround (if you do not care about the `Werkzeug security issue <https://github.com/predict-idlab/plotly-resampler/pull/174>`_): `pip install werkzeug==2.1.2`.
```

File: docs/sphinx/FAQ.rst [hunk 2]
```diff
@@ -66,6 +66,21 @@ The ``TraceUpdater`` class is a custom dash component that aids ``dcc.Graph`` co
...
+   </div>
+   </details>
+   <br>
+   <details>
+   <summary>
+      <a><b>My <code>FigureResampler.show_dash</code> keeps hanging (indefinitely) with the error message:<br>&nbsp;&nbsp;&nbsp; <code>OSError: Port already in use</code></b></a>
+   </summary>
+   <div style="margin-left:1em">
+
+Plotly-resampler its ``FigureResampler.show_dash`` method leverages the `jupyterdash <https://github.com/plotly/jupyter-dash>`_ toolkit to easily allow integration of dash apps in notebooks. However, there is a `known issue <https://github.com/plotly/jupyter-dash/pull/105>`_ with jupyterDash that causes the ``FigureResampler.show_dash`` method to hang when the port is already in use. In a future Pull-Request they will hopefully fix this issue.
```
Comment (jvdd) [inline docs/sphinx/FAQ.rst:80]:
```suggestion
Plotly-resampler its ``FigureResampler.show_dash`` method leverages the `jupyterdash <https://github.com/plotly/jupyter-dash>`_ toolkit to easily allow integration of dash apps in notebooks. However, there is a `known issue <https://github.com/plotly/jupyter-dash/pull/105>`_ with jupyterDash that causes the ``FigureResampler.show_dash`` method to hang when the port is already in use. In a future Pull-Request they will hopefully fix this issue. We internally track `this issue <https://github.com/predict-idlab/plotly-resampler/issues/123>` as well - please comment there if you want to provide feedback. 
```

**Comment by jvdd** [comment]
Eventually just removed the werkzeug dependency all together - this should default to the latest version.

**Pull request merged**